.github/actions/install_dbt_requirements/action.yaml

```diff
@@ -8,16 +8,25 @@ inputs:
 runs:
   using: composite
   steps:
-      - name: Setup python
-        uses: actions/setup-python@v4
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
         with:
-          python-version: '3.11'
-          cache: pip
+          enable-cache: true
+          cache-dependency-glob: |
+            ${{ inputs.dbt_project_dir }}/pyproject.toml
+            ${{ inputs.dbt_project_dir }}/uv.lock
 
-      - name: Install python requirements
-        working-directory: dbt
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version-file: ${{ inputs.dbt_project_dir }}/.python-version
+
+      - name: Install Python dependencies
+        working-directory: ${{ inputs.dbt_project_dir }}
         shell: bash
-        run: python -m pip install .
+        run: uv pip install .
+        env:
+          UV_SYSTEM_PYTHON: 1
```

Comment:
This env var [is required](https://docs.astral.sh/uv/guides/integration/github/#using-uv-pip) if we want to install the pip dependencies into the system library rather than a `uv` virtual env. We _could_ use the `uv` virtual env but we'd need to switch all the run commands from `python $SCRIPT` to `uv run $SCRIPT` and I thought this was easier.